src/net/larse/lcms/algorithms/VCT.java

```diff
@@ -345,79 +302,36 @@ public VCTOutput getResult(double[][] ud,
-//        this.ud[j][i] = ud.get(j).get(i);
-//      }
-//      this.mask[i] = mask.get(i);
-//      this.yearTable[i] = years.get(i);
-//    }
-//
-//    // Calculate the composite UD variable
-//    for (int i = 0; i < this.numYears; i++) {
-//      double sumSq = 0.0;
-//      for (Integer index : UD_INDEXES) {
-//        double tmp = this.ud[index][i];
-//        tmp = tmp >= 0.0 ? tmp : tmp / 2.5;
-//        sumSq += tmp * tmp;
-//      }
-//      this.ud[COMP][i] = Math.sqrt(sumSq / UD_INDEXES.size());
-//    }
-
-//      this.pctGoodObs = 0;
-//      this.longestDistLength = 0;
-//      this.longestDistR2 = 0.0;
-//      this.longestDistRough = 0.0;
-//      this.globalUdR2 = 0.0;
-//      this.globalUdSlp = 0.0;
-
-//      this.numMajorDist = 0;
-//    this.uRange = 0.0;
-//      this.vRange = 0.0;
-//      this.fiRough = 0.0;
-
-    //FIXME: (yang) do we need both uRange and fiRange, they seem to be the same in the code.
     this.fiRange = 0.0;
-//      this.hudSeg = 0;
-//      this.ludSeg = 0;
-//      this.agIndicator = 0;
-//      this.sharpTurns = 0;
-
-    //TODO: (yang) initialize variables, check for necessity!!
-    //which one can be local, not as class variables?
-    Arrays.fill(this.qFlag, 0);
-    Arrays.fill(this.cstSeg, 0);
-    Arrays.fill(this.cstSegSmooth, 0);
-    Arrays.fill(this.distFlag, 0);
-    Arrays.fill(this.distYear, 0);
-    Arrays.fill(this.distLength, 0);
-    Arrays.fill(this.regrFlag, 0);
-
-    Arrays.fill(this.distR2, 0);
-    Arrays.fill(this.distMagn, 0);
-    Arrays.fill(this.distMagnB4, 0);
-    Arrays.fill(this.distMagnVi, 0);
-    Arrays.fill(this.distMagnBr, 0);
-    Arrays.fill(this.regrR2, 0);
-    Arrays.fill(this.regrSlope, 0);
-    Arrays.fill(this.regrRough, 0);
-
-    Arrays.fill(this.meanForUdBx, 0);
-    Arrays.fill(this.sdForUdBx, 0);
+
+    // TODO: Keeping this as a range fill for now to be explicit that we
+    // would only want to fill up to this.numYears even if we decide to
+    // increase MAX_YEARS.  Overrule this if you want.
+    Arrays.fill(this.qFlag, 0, this.numYears, 0);
+    Arrays.fill(this.cstSeg, 0, this.numYears, 0);
+    Arrays.fill(this.cstSegSmooth, 0, this.numYears, 0);
+    Arrays.fill(this.distFlag, 0, this.numYears, 0);
+    Arrays.fill(this.distYear, 0, this.numYears, 0);
+    Arrays.fill(this.distLength, 0, this.numYears, 0);
+    Arrays.fill(this.regrFlag, 0, this.numYears, 0);
+
+    Arrays.fill(this.distR2, 0, this.numYears, 0.0);
+    Arrays.fill(this.distMagn, 0, this.numYears, 0.0);
+    Arrays.fill(this.distMagnB4, 0, this.numYears, 0.0);
+    Arrays.fill(this.distMagnVi, 0, this.numYears, 0.0);
+    Arrays.fill(this.distMagnBr, 0, this.numYears, 0.0);
+    Arrays.fill(this.regrR2, 0, this.numYears, 0.0);
+    Arrays.fill(this.regrSlope, 0, this.numYears, 0.0);
+    Arrays.fill(this.regrRough, 0, this.numYears, 0.0);
+
+    Arrays.fill(this.meanForUdBx, 0, N_BANDS, 0.0);
+    Arrays.fill(this.sdForUdBx, 0, N_BANDS, 0.0);
```

Comment:
Because we're now passing in `this.numYears` as the size the slice fill isn't necessary anymore.  Just left it here in case we go back to using `MAX_YEARS`